src/drivers/driver_zephyr.c

```diff
@@ -938,7 +956,10 @@ struct wpa_scan_results *wpa_drv_zep_get_scan_results2(void *priv)
 		goto out;
 	}
 
-	if_ctx->scan_res2 = os_zalloc(sizeof(*if_ctx->scan_res2));
+	if (!if_ctx->scan_res2) {
+		if_ctx->scan_res2 = os_zalloc(sizeof(*if_ctx->scan_res2));
```

Comment:
Can this be combined with the alloc check being done below ?

Reply:
This was intentional, if it's either non-NULL or allocation failure we need to catch that.